mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -267,14 +275,29 @@ impl TakerSavedSwap {
                 | TakerSwapEvent::MakerPaymentValidateFailed(_)
                 | TakerSwapEvent::TakerPaymentRefunded(_)
                 | TakerSwapEvent::TakerPaymentRefundedByWatcher(_)
-                | TakerSwapEvent::MakerPaymentSpent(_)
-                | TakerSwapEvent::MakerPaymentSpentByWatcher(_)
+                | TakerSwapEvent::MakerPaymentSpendConfirmed
                 | TakerSwapEvent::MakerPaymentWaitConfirmFailed(_) => {
                     return false;
                 },
+                TakerSwapEvent::MakerPaymentSpent(_) => {
+                    maker_payment_spent = true;
+                },
+                TakerSwapEvent::MakerPaymentSpentByWatcher(_) => {
+                    maker_payment_spent_by_watcher = true;
+                },
+                TakerSwapEvent::MakerPaymentSpendConfirmFailed(_) => {
+                    maker_payment_spend_confirmed_failed = true;
+                },
                 _ => (),
             }
         }
+        // MakerPaymentSpent or MakerPaymentSpentByWatcher were the last success events but a new step `MakerPaymentSpendConfirmed` was added after them.
+        // For backward compatibility (old saved swaps) we need to check for MakerPaymentSpent or MakerPaymentSpentByWatcher
+        // and if there is no MakerPaymentSpendConfirmFailed.
+        if (maker_payment_spent || maker_payment_spent_by_watcher) && !maker_payment_spend_confirmed_failed {
+            return false;
+        }
+
         true
```

Comment:
can be simplified to
```rust
maker_payment_spend_confirmed_failed || (!maker_payment_spent && !maker_payment_spent_by_watcher) 
```

Reply:
But this leads to another logic, its not the same as this if check
```
(maker_payment_spent || maker_payment_spent_by_watcher) && !maker_payment_spend_confirmed_failed
```

Reply:
I got what you mean :) https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/commits/b33380aead363d7c4088bf5812f35a317b0586a6